### PR TITLE
New: Attaches comments to the ESTree AST (fixes #31)

### LINKED ISSUE #31: Phase 3: Attach comments
Phase 3 is to add attaching leading/trailing comments to the AST inside the parser. This would involve:

1. Retrieving comments for each node.
1. Converting the TypeScript comments into Esprima-style comments.
1. Adding the comments into `leadingComments` and `trailingComments` arrays on appropriate nodes.
1. Making the result pass the tests that are already tests in tests/fixtures/attach-comments.

### PR DESCRIPTION
**Pull request by JamesHenry**
See https://github.com/eslint/typescript-eslint-parser/pull/34 for all the discussion leading up to this implementation

### CODE REVIEW

File: lib/ast-converter.js [hunk 1]
```diff
@@ -268,6 +267,8 @@ function getTokenType(token) {
 
             case SyntaxKind.GetKeyword:
             case SyntaxKind.SetKeyword:
+            case SyntaxKind.TypeKeyword:
```
Comment (nzakas) [inline lib/ast-converter.js]: It looks like there are a bunch of unrelated changes in this file?

Comment (JamesHenry) [inline lib/ast-converter.js]: They surfaced as bugs/missing checks when implementing the original work on this branch. I left them in because they seem like bugs waiting to happen, but I am happy to remove them if you feel that would be cleaner.

**Comment by nzakas** [comment]
As noted inline, it looks like there are a lot of unrelated changes in this PR. Can you narrow it down so the changes are only related to finding the comments?

**Comment by JamesHenry** [comment]
It is not as simple as saying they are unrelated to finding the comments... For example:
```
case SyntaxKind.TrueKeyword:
                assign(result, {
                    type: "Literal",
                    value: true,
                    raw: "true" // <- This was added in the PR
                });
                break;
```

The addition of the raw values for TrueKeyword, FalseKeyword and NullKeyword is necessary to make the attach-comments tests pass.

**Comment by nzakas** [comment]
Oh I see, thanks for explaining. In that case, I think this is good to go.

**Comment by JamesHenry** [comment]
Great, thanks!
